lib/names/hostname.go

```diff
@@ -1,16 +1,13 @@
 package names
 
-import (
-	"os"
-	"strings"
-)
+import "os"
 
 // Hostname returns this hosts hostname, converting to lowercase so that
 // it is valid for use in the Calico API.
 func Hostname() (string, error) {
 	if h, err := os.Hostname(); err != nil {
 		return "", err
 	} else {
-		return strings.ToLower(strings.TrimSpace(h)), nil
```

Comment:
Wonder if this code actually needs to do smart truncation of FQDNs that are longer that the maximum name length.